cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -290,6 +262,65 @@ public ScheduleState scheduleState(Id.Program program, SchedulableProgramType pr
     }
   }
 
+  private void schedule(Id.Program program, ScheduleSpecification scheduleSpec, Map<String, String> properties,
+                       boolean updateMds) throws SchedulerException {
+    if (updateMds) {
+      store.addSchedule(program, scheduleSpec);
+    }
+    Schedule schedule = scheduleSpec.getSchedule();
+    Scheduler scheduler;
+    if (schedule instanceof TimeSchedule) {
+      scheduler = timeScheduler;
+    } else if (schedule instanceof StreamSizeSchedule) {
+      scheduler = streamSizeScheduler;
+    } else {
+      throw new IllegalArgumentException("Unhandled type of schedule: " + schedule.getClass());
+    }
+
+    SchedulableProgramType programType = program.getType().getSchedulableType();
+    scheduler.schedule(program, programType, ImmutableList.of(schedule), properties);
+    if (isLazyStart()) {
+      // TODO: CDAP-2281 figure out a better way to handle schedules in unit tests
+      String ignoreLazy = properties.get(Constants.Scheduler.IGNORE_LAZY_START);
+      boolean shouldNotSuspend = ignoreLazy != null && Boolean.valueOf(ignoreLazy);
+      if (shouldNotSuspend) {
+        // normally in lazy mode, the scheduler is started on calls to resume.
+        // If this schedule should be active right now instead of requiring a call to resume,
+        // we need to start the scheduler here.
+        lazyStart(scheduler);
+      } else {
+        try {
+          scheduler.suspendSchedule(program, programType, schedule.getName());
+        } catch (NotFoundException e) {
+          // Should not happen - we just created it. Could have been deleted just in between
+          LOG.info("Schedule could not be suspended - it did not exist: {}", schedule.getName());
+        }
+      }
+    }
+  }
+
+  private void deleteSchedules(Id.Program program, SchedulableProgramType programType,
+                               boolean deleteFromStore) throws SchedulerException {
```

Comment:
Looks like the only place where we would want to do it is from `deleteAllSchedules